# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/dav1d/src/arm/64/util.S
# Commit: a5108ffbd311
# Full Hash: a5108ffbd31101e7cb72dabc271d92027f953c3c
# Author: Alex Chronopoulos <achronop@gmail.com>
# Date: 2019-04-20 09:35:22
# Regressor Bug: 1540830
# File Overlap Count: 3
# Description:
#   Bug 1540830 - Update dav1d from upstream to 1f7a7e8. r=TD-Linux
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D28200
# ==============================================================================

diff -r 98adabf295d0 -r a5108ffbd311 third_party/dav1d/src/arm/64/util.S
--- a/third_party/dav1d/src/arm/64/util.S	Fri Apr 19 20:49:36 2019 +0000
+++ b/third_party/dav1d/src/arm/64/util.S	Fri Apr 19 20:36:10 2019 +0000
@@ -35,57 +35,98 @@
 .macro  movrel rd, val, offset=0
 #if defined(__APPLE__)
   .if \offset < 0
-    adrp        \rd, \val@PAGE
-    add         \rd, \rd, \val@PAGEOFF
-    sub         \rd, \rd, -(\offset)
+        adrp            \rd, \val@PAGE
+        add             \rd, \rd, \val@PAGEOFF
+        sub             \rd, \rd, -(\offset)
   .else
-    adrp        \rd, \val+(\offset)@PAGE
-    add         \rd, \rd, \val+(\offset)@PAGEOFF
+        adrp            \rd, \val+(\offset)@PAGE
+        add             \rd, \rd, \val+(\offset)@PAGEOFF
   .endif
 #elif defined(PIC) && defined(_WIN32)
   .if \offset < 0
-    adrp        \rd, \val
-    add         \rd, \rd, :lo12:\val
-    sub         \rd, \rd, -(\offset)
+        adrp            \rd, \val
+        add             \rd, \rd, :lo12:\val
+        sub             \rd, \rd, -(\offset)
   .else
-    adrp        \rd, \val+(\offset)
-    add         \rd, \rd, :lo12:\val+(\offset)
+        adrp            \rd, \val+(\offset)
+        add             \rd, \rd, :lo12:\val+(\offset)
   .endif
 #elif defined(PIC)
-    adrp        \rd, \val+(\offset)
-    add         \rd, \rd, :lo12:\val+(\offset)
+        adrp            \rd, \val+(\offset)
+        add             \rd, \rd, :lo12:\val+(\offset)
 #else
-    ldr         \rd, =\val+\offset
+        ldr             \rd, =\val+\offset
 #endif
 .endm
 
 .macro transpose_8x8b r0, r1, r2, r3, r4, r5, r6, r7, r8, r9
-    trn1        \r8\().8b,  \r0\().8b,  \r1\().8b
-    trn2        \r9\().8b,  \r0\().8b,  \r1\().8b
-    trn1        \r1\().8b,  \r2\().8b,  \r3\().8b
-    trn2        \r3\().8b,  \r2\().8b,  \r3\().8b
-    trn1        \r0\().8b,  \r4\().8b,  \r5\().8b
-    trn2        \r5\().8b,  \r4\().8b,  \r5\().8b
-    trn1        \r2\().8b,  \r6\().8b,  \r7\().8b
-    trn2        \r7\().8b,  \r6\().8b,  \r7\().8b
+        trn1            \r8\().8b,  \r0\().8b,  \r1\().8b
+        trn2            \r9\().8b,  \r0\().8b,  \r1\().8b
+        trn1            \r1\().8b,  \r2\().8b,  \r3\().8b
+        trn2            \r3\().8b,  \r2\().8b,  \r3\().8b
+        trn1            \r0\().8b,  \r4\().8b,  \r5\().8b
+        trn2            \r5\().8b,  \r4\().8b,  \r5\().8b
+        trn1            \r2\().8b,  \r6\().8b,  \r7\().8b
+        trn2            \r7\().8b,  \r6\().8b,  \r7\().8b
+
+        trn1            \r4\().4h,  \r0\().4h,  \r2\().4h
+        trn2            \r2\().4h,  \r0\().4h,  \r2\().4h
+        trn1            \r6\().4h,  \r5\().4h,  \r7\().4h
+        trn2            \r7\().4h,  \r5\().4h,  \r7\().4h
+        trn1            \r5\().4h,  \r9\().4h,  \r3\().4h
+        trn2            \r9\().4h,  \r9\().4h,  \r3\().4h
+        trn1            \r3\().4h,  \r8\().4h,  \r1\().4h
+        trn2            \r8\().4h,  \r8\().4h,  \r1\().4h
+
+        trn1            \r0\().2s,  \r3\().2s,  \r4\().2s
+        trn2            \r4\().2s,  \r3\().2s,  \r4\().2s
+        trn1            \r1\().2s,  \r5\().2s,  \r6\().2s
+        trn2            \r5\().2s,  \r5\().2s,  \r6\().2s
+        trn2            \r6\().2s,  \r8\().2s,  \r2\().2s
+        trn1            \r2\().2s,  \r8\().2s,  \r2\().2s
+        trn1            \r3\().2s,  \r9\().2s,  \r7\().2s
+        trn2            \r7\().2s,  \r9\().2s,  \r7\().2s
+.endm
 
-    trn1        \r4\().4h,  \r0\().4h,  \r2\().4h
-    trn2        \r2\().4h,  \r0\().4h,  \r2\().4h
-    trn1        \r6\().4h,  \r5\().4h,  \r7\().4h
-    trn2        \r7\().4h,  \r5\().4h,  \r7\().4h
-    trn1        \r5\().4h,  \r9\().4h,  \r3\().4h
-    trn2        \r9\().4h,  \r9\().4h,  \r3\().4h
-    trn1        \r3\().4h,  \r8\().4h,  \r1\().4h
-    trn2        \r8\().4h,  \r8\().4h,  \r1\().4h
+.macro transpose_8x16b r0, r1, r2, r3, r4, r5, r6, r7, r8, r9
+        trn1            \r8\().16b, \r0\().16b, \r1\().16b
+        trn2            \r9\().16b, \r0\().16b, \r1\().16b
+        trn1            \r1\().16b, \r2\().16b, \r3\().16b
+        trn2            \r3\().16b, \r2\().16b, \r3\().16b
+        trn1            \r0\().16b, \r4\().16b, \r5\().16b
+        trn2            \r5\().16b, \r4\().16b, \r5\().16b
+        trn1            \r2\().16b, \r6\().16b, \r7\().16b
+        trn2            \r7\().16b, \r6\().16b, \r7\().16b
+
+        trn1            \r4\().8h,  \r0\().8h,  \r2\().8h
+        trn2            \r2\().8h,  \r0\().8h,  \r2\().8h
+        trn1            \r6\().8h,  \r5\().8h,  \r7\().8h
+        trn2            \r7\().8h,  \r5\().8h,  \r7\().8h
+        trn1            \r5\().8h,  \r9\().8h,  \r3\().8h
+        trn2            \r9\().8h,  \r9\().8h,  \r3\().8h
+        trn1            \r3\().8h,  \r8\().8h,  \r1\().8h
+        trn2            \r8\().8h,  \r8\().8h,  \r1\().8h
 
-    trn1        \r0\().2s,  \r3\().2s,  \r4\().2s
-    trn2        \r4\().2s,  \r3\().2s,  \r4\().2s
-    trn1        \r1\().2s,  \r5\().2s,  \r6\().2s
-    trn2        \r5\().2s,  \r5\().2s,  \r6\().2s
-    trn2        \r6\().2s,  \r8\().2s,  \r2\().2s
-    trn1        \r2\().2s,  \r8\().2s,  \r2\().2s
-    trn1        \r3\().2s,  \r9\().2s,  \r7\().2s
-    trn2        \r7\().2s,  \r9\().2s,  \r7\().2s
+        trn1            \r0\().4s,  \r3\().4s,  \r4\().4s
+        trn2            \r4\().4s,  \r3\().4s,  \r4\().4s
+        trn1            \r1\().4s,  \r5\().4s,  \r6\().4s
+        trn2            \r5\().4s,  \r5\().4s,  \r6\().4s
+        trn2            \r6\().4s,  \r8\().4s,  \r2\().4s
+        trn1            \r2\().4s,  \r8\().4s,  \r2\().4s
+        trn1            \r3\().4s,  \r9\().4s,  \r7\().4s
+        trn2            \r7\().4s,  \r9\().4s,  \r7\().4s
+.endm
+
+.macro  transpose_4x16b r0, r1, r2, r3, t4, t5, t6, t7
+        trn1            \t4\().16b, \r0\().16b, \r1\().16b
+        trn2            \t5\().16b, \r0\().16b, \r1\().16b
+        trn1            \t6\().16b, \r2\().16b, \r3\().16b
+        trn2            \t7\().16b, \r2\().16b, \r3\().16b
+
+        trn1            \r0\().8h,  \t4\().8h,  \t6\().8h
+        trn2            \r2\().8h,  \t4\().8h,  \t6\().8h
+        trn1            \r1\().8h,  \t5\().8h,  \t7\().8h
+        trn2            \r3\().8h,  \t5\().8h,  \t7\().8h
 .endm
 
 #endif /* DAV1D_SRC_ARM_64_UTIL_S */